packages/perpswap/src/contracts/vault.rs

```diff
@@ -12,21 +13,21 @@ pub struct InstantiateMsg {
     pub governance: String,
 
     /// Initial allocation percentages to markets
-    pub markets_allocation_bps: Vec<u16>,
+    pub markets_allocation_bps: HashMap<String, u16>,
```

Comment:
Is the key here a market ID? If so, can you use the `MarketId` data type?

Reply:
I'm OK with market address. If you do that, please use the `Addr` datatype instead.